src/inmanta/data/__init__.py

```diff
@@ -4815,6 +4815,299 @@ def get_compliance_status(self) -> Optional[state.Compliance]:
             return state.Compliance.NON_COMPLIANT
 
 
+class ResourceSet(BaseDocument):
+    """
+    A set of resources
+
+    :param environment: The environment this resource set belongs to
+    :param id: The id of this resource set. Unique per environment.
+    :param name: The name of this resource set, None if it is the default set
+    """
+
+    environment: uuid.UUID
+    id: uuid.UUID
+    name: Optional[str]
+
+    @classmethod
+    def table_name(cls) -> str:
+        return "resource_set"
+
+    @classmethod
+    async def get_resource_sets_in_version(
+        cls, environment: uuid.UUID, version: int, connection: Optional[asyncpg.connection.Connection] = None
+    ) -> list["ResourceSet"]:
+        """
+        Returns the resource sets in the given version. Only meant for testing.
+        """
+        query = f"""
+            SELECT rs.*
+            FROM public.resource_set_configuration_model rscm
+            INNER JOIN {cls.table_name()} rs
+                ON rs.id=rscm.resource_set_id
+            WHERE rscm.environment=$1 AND rscm.model=$2
+                """
+        query_result = await cls._fetch_query(
+            query,
+            environment,
+            version,
+            connection=connection,
+        )
+        result = [cls(from_postgres=True, **record) for record in query_result]
+        # Could not express this constraint in the database
+        assert len({rs.name for rs in result}) == len(
+            result
+        ), "Inconsistency in the database, a resource set cannot be present more than once in the same model version"
+        return result
+
+    @classmethod
+    def get_printable_name_for_resource_set(cls, native: str | None) -> str:
+        if native is None:
+            return "<SHARED>"
+        return native
+
+    @classmethod
+    async def validate_resource_sets_in_version(
+        cls,
+        environment: uuid.UUID,
+        version: int,
+        updated_resource_sets: list[str | None],
+        connection: Optional[asyncpg.connection.Connection] = None,
```

Comment:
Probably doesn't make sense to make connection optional here. Also, make it a kw-only arg.